chore: ratchet-down complexity thresholds

## Automated ratchet-down (t1913)

Simplification wins have accumulated. The following thresholds can be lowered:

```
BASH32_COMPAT_THRESHOLD: 78 → 74 (actual: 72, gap: 6)
```

### Worker Guidance

**Files to Modify:**
- `EDIT: .agents/configs/complexity-thresholds.conf` — lower the thresholds listed above

**Do NOT modify:**
- `.agents/configs/simplification-state.json` — this is a maintenance registry updated by the pulse on `main` only. Including it in a focused ratchet-down PR causes diff noise across hundreds of unrelated entries and increases merge-conflict surface (GH#18622). The pulse refreshes it automatically after the PR merges. If it appears modified in your worktree, do not `git add` it.

**Implementation Steps:**
1. For each proposed threshold, update the value in `complexity-thresholds.conf`
2. Add a ratchet-down comment above the updated value documenting the change (e.g., `# Ratcheted down to NNN (GH#NNNN): actual violations NNN + 2 buffer`)
3. Do NOT remove existing bump history comments — they are the audit trail
4. Verify the state file is NOT staged before committing: `git diff --cached --name-only | grep -v simplification-state`

**Verification:**
```bash
# Confirm thresholds updated
grep -E 'FUNCTION_COMPLEXITY_THRESHOLD|NESTING_DEPTH_THRESHOLD|FILE_SIZE_THRESHOLD|BASH32_COMPAT_THRESHOLD' .agents/configs/complexity-thresholds.conf
# Confirm CI would pass with new values
.agents/scripts/complexity-scan-helper.sh ratchet-check . 5
# Confirm state file is NOT staged in the PR commit
git diff --cached --name-only | grep -v 'simplification-state.json'
```

<!-- aidevops:generator=ratchet-down -->
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.8.65 automated scan.

> Premise falsified. The ratchet-down scanner reported 72 local violations, but CI consistently reports 76. Threshold stays at 78.

**Triage outcome:** CI environment has a persistent 4-violation gap vs local scan for `BASH32_COMPAT_THRESHOLD`.

**Evidence (CI run 24557873248, job 71798804688):**
```
Bash 3.2 compatibility violations: 76
##[error]Bash 3.2 compatibility regression: 76 violations (threshold: 74)
```

**Pattern:** This same failure has occurred on every BASH32 ratchet attempt from GH#19423 through GH#19547 — 10+ consecutive attempts, all showing local=72 but CI=76.

**PR #19553** adds an audit-trail comment to `complexity-thresholds.conf` documenting this attempt and keeps the threshold at 78. All CI checks on PR #19553 pass (threshold restored to 78).

**Root cause hypothesis:** The CI file scanner includes 4 additional files that the local complexity-scan-helper.sh excludes. Not acting on this further — it's a systemic discrepancy that needs investigation into why local and CI scan different file sets for the bash32 check.

--- a/.agents/configs/complexity-thresholds.conf
+++ b/.agents/configs/complexity-thresholds.conf
@@ -269,7 +269,8 @@ FILE_SIZE_THRESHOLD=59
 # Local scan saw 72 violations but CI runner sees 76. 76 + 2 buffer = 78.
 # GH#19547 attempted ratchet to 74 (claimed actual: 72), but CI reported 76 violations
 # against threshold 74 — same drift pattern as all prior attempts. Keeping at 78.
-BASH32_COMPAT_THRESHOLD=78
+# Ratcheted down to 74 (GH#19551): actual violations 72 + 2 buffer
+BASH32_COMPAT_THRESHOLD=74
 
 # Qlty maintainability smell baseline (t2065, GH#18773). Seed value for
 # the `.github/workflows/qlty-regression.yml` gate. The gate itself uses

--- a/.agents/configs/complexity-thresholds.conf
+++ b/.agents/configs/complexity-thresholds.conf
@@ -269,8 +269,9 @@ FILE_SIZE_THRESHOLD=59
 # Local scan saw 72 violations but CI runner sees 76. 76 + 2 buffer = 78.
 # GH#19547 attempted ratchet to 74 (claimed actual: 72), but CI reported 76 violations
 # against threshold 74 — same drift pattern as all prior attempts. Keeping at 78.
-# Ratcheted down to 74 (GH#19551): actual violations 72 + 2 buffer
-BASH32_COMPAT_THRESHOLD=74
+# GH#19551 attempted ratchet to 74 (claimed actual: 72), but CI reported 76 violations
+# against threshold 74 — same drift pattern as all prior attempts. Keeping at 78.
+BASH32_COMPAT_THRESHOLD=78
 
 # Qlty maintainability smell baseline (t2065, GH#18773). Seed value for
 # the `.github/workflows/qlty-regression.yml` gate. The gate itself uses

